garrysmod/lua/includes/modules/list.lua

```diff
@@ -1,90 +1,56 @@
---=============================================================================--							 
---
---  A really simple module to allow easy additions to lists of items
---
---=============================================================================--
-
-local table 	= table
 local pairs		= pairs
+local table 	= table
 
-module( "list" )
-
-
-local g_Lists = {}
 
---
---	Get a list
---
-function Get( list )
+module( "list" )
 
-	g_Lists[ list ] = g_Lists[ list ] or {}
-	return table.Copy( g_Lists[ list ] )
-	
-end
 
---
---	Get a list
---
-function GetForEdit( list )
+local Lists = {}
 
-	g_Lists[ list ] = g_Lists[ list ] or {}
-	return g_Lists[ list ]
-	
+function GetStored( listid )
+	return Lists[ listid ]
 end
 
---
---	Get all list names
---
-function GetTable()
+function GetForEdit( listid )
+	local list = Lists[ listid ]
 
-	return table.GetKeys( g_Lists )
+	if ( list == nil ) then
+		list = {}
+		Lists[ listid ] = list
+	end
 
+	return list
 end
 
---
---	Set a key value
---
-function Set( list, key, value )
-
-	local list = GetForEdit( list )
-	list[ key ] = value
-
+function Get( listid )
+	return table.Copy( GetForEdit( listid ) )
 end
 
---
---	Add a value to a list
---
-function Add( list, value )
-
-	local list = GetForEdit( list )
-	return table.insert( list, value )
-
+function GetTable()
+	return table.GetKeys( Lists )
 end
 
---
---	Returns true if the list contains the value (as a value - not a key)
---
-function Contains( list, value )
-
-	if ( !g_Lists[ list ] ) then return false end
+function Set( listid, key, val )

```

Comment:
How is `val` is "better var naming" than `value`? There's no good reason for this change as far as I can tell

It's shorter? Sure. Why does it need to be 2 characters shorter?

Reply:
For naming consistency. ``key, val`` is the standard arg naming for most Lua codebases I've come across (and ``k, v`` for loops). ``val`` ( https://github.com/search?q=repo%3AFacepunch%2Fgarrysmod+val+language%3ALua&type=code&l=Lua ) is used more than ``value`` ( https://github.com/search?q=repo%3AFacepunch%2Fgarrysmod+value+language%3ALua&type=code ) in the gmod codebase currently (rough search but val seems to be the most used for function args). This doesn't really "matter" but the goal of these file-level refactor PRs is to set a consistent and optimal standard of practices for the entire codebases that devs can learn from, and common arg naming is just an extension of that.

Reply:
This is an unbelievably silly argument. Not only that, your example doesn't really work if you do it more properly:

![image](https://github.com/Facepunch/garrysmod/assets/3299036/f3df6e19-a89d-4971-9ff0-a22e02c46c48)
![image](https://github.com/Facepunch/garrysmod/assets/3299036/1f0481d8-1a18-4e38-8a19-397ec4f8d153)

(This is across the GMod repo, granted the word `value` appears in some comments as well)

This feels more of a you forcing your own preferences into the repo.

Reply:
I'd also would like to invite you to read https://github.com/Facepunch/garrysmod/blob/master/CONTRIBUTING.md

Reply:
I highly disagree that's it's "silly" since it's introducing consistency though I'll concede that it's pedantic. If it will get the review thru, I'll revert it.

Reply:
Also, value is inflated in searches because of member variables which (correctly) use the full name, but alas.